i3lock.c

```diff
@@ -1065,4 +1087,17 @@ int main(int argc, char *argv[]) {
      * file descriptor becomes readable). */
     ev_invoke(main_loop, xcb_check, 0);
     ev_loop(main_loop, 0);
+
+    if (stolen_focus == XCB_NONE) {
+        return 0;
+    }
+
+    DEBUG("restoring focus to X11 window 0x%08x\n", stolen_focus);
+    xcb_ungrab_pointer(conn, XCB_CURRENT_TIME);
+    xcb_ungrab_keyboard(conn, XCB_CURRENT_TIME);
+    xcb_destroy_window(conn, win);
+    set_focused_window(conn, screen->root, stolen_focus);
```

Comment:
Why don't we use SetInputFocus here as well? We're already using it above. What if prior to locking the focus belonged to an unmanaged window? We are simply restoring focus to the wrong window here.

Reply:
> Why don't we use SetInputFocus here as well? We're already using it above.

Because _NET_ACTIVE_WINDOW is the better option: it is a request to the window manager, whereas SetInputFocus circumvents the window manager. The fact that we must resort to SetInputFocus for the part where focus concerns an unmanaged window doesn’t mean we have to give up on _NET_ACTIVE_WINDOW in the other cases.

> What if prior to locking the focus belonged to an unmanaged window? We are simply restoring focus to the wrong window here.

If an unmanaged window was focused, a different window will be focused, yes. i3lock, without my proposed change, already screws up focus when the pointer is over a different window, so we’re not making things worse: we’re fixing a somewhat common case at the expense of a much rarer case:

Unmanaged windows are quite the edge case. Given that i3 in general prioritizes tiling windows and considers floating windows a nuisance, I think it is fair to say that we don’t need to care about focus restore behavior for unmanaged windows when unlocking your screen.